engine/src/flutter/lib/ui/geometry.dart

```diff
@@ -1806,21 +1827,25 @@ class RRect extends _RRectLike<RRect> {
 
 /// An immutable rounded superellipse.
 ///
-/// A rounded superellipse is a shape similar to a typical rounded rectangle
-/// ([RRect]), but with smoother transitions between the straight sides and the
-/// rounded corners. It resembles the `RoundedRectangle` shape in SwiftUI with
-/// the `.continuous` corner style.
+/// A rounded superellipse (not to be confused with a standard superellipse) is
+/// a shape formed by replacing the four curved corners of a superellipse with
+/// circular arcs. A superellipse follows the formula x^n + y^n = a^n, and while
+/// n > 2 gives it rounded corners, they tend to be too sharp and pronounced.
+/// Replacing them with circular arcs makes the shape feel softer and more
+/// natural.
 ///
-/// Technically, a canonical rounded superellipse, i.e. one with a uniform
-/// corner radius ([RSuperellipse.fromRectAndRadius]), is created by replacing
-/// the four corners of a superellipse (also known as a Lamé curve) with
-/// circular arcs. A rounded superellipse with non-uniform radii is extended on it
-/// by concatenating arc segments and transformation.
+/// Visually, a rounded superellipse looks similar to a typical rounded rectangle
+/// ([RRect]) but with smoother transitions between the straight edges and
+/// corners. It closely matches the `RoundedRectangle` shape in SwiftUI with the
+/// `.continuous` corner style.
 ///
-/// The corner radius parameters used in this class corresponds to SwiftUI's
-/// `cornerRadius` parameter, which is close to, but not exactly equals to, the
-/// radius of the corner circles.
-class RSuperellipse extends _RRectLike<RSuperellipse> {
+/// The [RSuperellipse] class is `const`-constructible, making it efficiently
+/// comparable. However, this also means that computations must start from
+/// scratch each time they are performed. The [computed] method returns an
+/// instance that can cache intermediate values, improving performance. For
+/// efficiency, it is recommended to reuse and cache these instances whenever
+/// possible.
+class RSuperellipse with _RRectLike<RSuperellipse> {
```

Comment:
I would make this final, as the native code relies on casting the results of computed to a private type, any other implementations would fail if used by the canvas.

Reply:
I've changed the design, and now the native class is a separate class that _represents_ the `param` instead of a subclass of `RSuperellipse`. I think that renders this comment obsolete.